packages/yew/src/functional/hooks/use_state.rs

```diff
@@ -149,6 +159,15 @@ where
     }
 }
 
+impl<T> PartialEq<T> for UseStateHandle<T>
+where
+    T: PartialEq,
+{
+    fn eq(&self, rhs: &T) -> bool {
+        &self.inner.value == rhs
```

Comment:
We have the following implementation on `UseStateSetter<T>` (a setter only variant to `UseStateHandle<T>`) where it asserts whether 2 setters are setting into the same state:

```rust
impl<T> PartialEq for UseStateSetter<T> {
    fn eq(&self, rhs: &Self) -> bool {
        self.inner == rhs.inner
    }
}
```

I think implementing `UseStateSetter<T>` to test whether 2 setters are setting into the same state and `UseStateHandle<T>` to compare the value might be some what confusing to users.

I feel it might be better to use `left_handle.as_ref() == right_handle.as_ref()` in this case.

Reply:
Agreed. This is comparing two handles, not the values of those handles